File: engine/src/flutter/shell/platform/android/android_context.cc
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "flutter/shell/platform/android/android_context.h"

#include "flutter/shell/platform/android/android_context_gl.h"
#include "flutter/shell/platform/android/android_environment_gl.h"

namespace flutter {

std::shared_ptr<AndroidContext> CreateContextGL() {
  auto context_gl = std::make_shared<AndroidContextGL>(
      AndroidRenderingAPI::kOpenGLES,
      fml::MakeRefCounted<AndroidEnvironmentGL>());
  FML_CHECK(context_gl->IsValid()) << "Could not create an Android context GL.";
  return context_gl;
}

AndroidContext::AndroidContext(AndroidRenderingAPI rendering_api)
    : rendering_api_(rendering_api) {}

AndroidContext::~AndroidContext() = default;

AndroidRenderingAPI AndroidContext::RenderingApi() const {
  return rendering_api_;
}

std::shared_ptr<AndroidContext> AndroidContext::Create(
    AndroidRenderingAPI rendering_api) {
  if (rendering_api == AndroidRenderingAPI::kOpenGLES) {
    return CreateContextGL();
  }
  return std::make_shared<AndroidContext>(rendering_api);
}

}  // namespace flutter

File: packages/flutter_tools/lib/src/context/android_context.dart
// Copyright 2014 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import '../android/android_sdk.dart';
import '../android/android_studio.dart';
import '../android/gradle_utils.dart';
import '../android/java.dart';

/// Holds Android-specific dependencies.
class AndroidContext {
  AndroidContext({
    required this.androidSdkBuilder,
    required this.androidStudioBuilder,
    required this.gradleUtilsBuilder,
    required this.javaBuilder,
  });

  /// Discovers, validates, and manages the local Android SDK and platform tools.
  late final AndroidSdk? androidSdk = androidSdkBuilder();

  /// Discovers and inspects local Android Studio installations and embedded JDK paths.
  late final AndroidStudio? androidStudio = androidStudioBuilder();

  /// Utility helpers for interacting with Gradle builds and resolving project configs.
  late final GradleUtils gradleUtils = gradleUtilsBuilder();

  /// Discovers and validates the active Java Development Kit (JDK) binary.
  late final Java? java = javaBuilder();

  final AndroidSdk? Function() androidSdkBuilder;
  final AndroidStudio? Function() androidStudioBuilder;
  final GradleUtils Function() gradleUtilsBuilder;
  final Java? Function() javaBuilder;
}
